DataManagementSystem/scripts/dirac-dms-replicate-and-register-request.py

```diff
@@ -52,54 +49,71 @@ def getLFNList( LFNs ):
-  replicateAndRegister.TargetSE = ",".join( targetSEs )
-
-  for lfn in lfnList:
-    metaDict = metaDatas["Successful"][lfn]
-    opFile = File()
-    opFile.LFN = lfn
-    opFile.Size = metaDict["Size"]
-
-    if "Checksum" in metaDict:
-      # # should check checksum type, now assuming Adler32 (metaDict["ChecksumType"] = 'AD'
-      opFile.Checksum = metaDict["Checksum"]
-      opFile.ChecksumType = "ADLER32"
-    replicateAndRegister.addFile( opFile )
-
-  request.addOperation( replicateAndRegister )
+  lfnChunks = breakListIntoChunks( lfnList, 100 )
+  multiRequests = len( lfnChunks ) > 1
 
+  error = 0
+  count = 0
   reqClient = ReqClient()
-  putRequest = reqClient.putRequest( request )
-  if not putRequest["OK"]:
-    gLogger.error( "unable to put request '%s': %s" % ( requestName, putRequest["Message"] ) )
-    DIRAC.exit( -1 )
-
-  gLogger.always( "Request '%s' has been put to ReqDB for execution." % requestName )
-  gLogger.always( "You can monitor its status using command: 'dirac-rms-show-request %s'" % requestName )
-  DIRAC.exit( 0 )
+  fc = FileCatalog()
+  for lfnChunk in lfnChunks:
+    metaDatas = fc.getFileMetadata( lfnChunk )
+    if not metaDatas["OK"]:
+      gLogger.error( "unable to read metadata for lfns: %s" % metaDatas["Message"] )
+      error = -1
+      continue
+    metaDatas = metaDatas["Value"]
+    for failedLFN, reason in metaDatas["Failed"].items():
+      gLogger.error( "skipping %s: %s" % ( failedLFN, reason ) )
+    lfnChunk = set( metaDatas["Successful"] )
+
+    if not lfnChunk:
+      gLogger.error( "LFN list is empty!!!" )
+      error = -1
+      continue
+
+    if len( lfnChunk ) > Operation.MAX_FILES:
+      gLogger.error( "too many LFNs, max number of files per operation is %s" % Operation.MAX_FILES )
+      error = -1
+      continue
+
+    count += 1
+    request = Request()
+    request.RequestName = requestName if not multiRequests else '%s_%d' % ( requestName, count )
+
+    replicateAndRegister = Operation()
+    replicateAndRegister.Type = "ReplicateAndRegister"
+    replicateAndRegister.TargetSE = ",".join( targetSEs )
+
+    for lfn in lfnChunk:
+      metaDict = metaDatas["Successful"][lfn]
+      opFile = File()
+      opFile.LFN = lfn
+      opFile.Size = metaDict["Size"]
+
+      if "Checksum" in metaDict:
+        # # should check checksum type, now assuming Adler32 (metaDict["ChecksumType"] = 'AD'
+        opFile.Checksum = metaDict["Checksum"]
+        opFile.ChecksumType = "ADLER32"
+      replicateAndRegister.addFile( opFile )
+
+    request.addOperation( replicateAndRegister )
+
+    putRequest = reqClient.putRequest( request )
+    if not putRequest["OK"]:
+      gLogger.error( "unable to put request '%s': %s" % ( request.RequestName, putRequest["Message"] ) )
+      error = -1
+      continue
+
+    if not multiRequests:
+      gLogger.always( "Request '%s' has been put to ReqDB for execution." % request.RequestName )
+
+  if multiRequests:
```

Comment:
The show request command can be versatile like for example dsub command ( COMDIRAC extension ) showing all the requests in a nice table, one row per request, with various selection switches. By default, showing all the requests of the current user/group. With verbose mode it can show more details for selected requests. Similarity of job and request look-up commands will be certainly an advantage for non-experienced users